nixos/modules/tasks/network-interfaces-scripted.nix

```diff
@@ -180,9 +180,9 @@ in
           { description = "Virtual Network Interface ${i.name}";
             bindsTo = [ "dev-net-tun.device" ];
             after = [ "dev-net-tun.device" "network-pre.target" ];
-            wantedBy = [ "network-setup.service" (subsystemDevice i.name) ];
+            wantedBy = [ "network-setup.service" "network-addresses-${i.name}.service" "network-link-${i.name}.service" (subsystemDevice i.name) ];
```

Comment:
Can you explain what problem you solve by adding these units to `wantedBy` and `before`?